packages/esm-appointments-app/src/helpers/time.tsx

```diff
@@ -19,23 +19,23 @@ export const convertTime12to24 = (time12h, timeFormat: amPm) => {
   return [hours, minutes];
 };
 
-const initialState = { appointmentDate: dayjs(new Date().setHours(0, 0, 0, 0)).format(omrsDateFormat) };
+const initialState = { selectedDate: dayjs(new Date().setHours(0, 0, 0, 0)).format(omrsDateFormat) };
 
-export function getStartDate() {
-  return getGlobalStore<{ appointmentDate: string | Date }>('appointmentStartDate', initialState);
+export function getSelectedDate() {
+  return getGlobalStore<{ selectedDate: string }>('appointment-selected-date', initialState);
 }
 
-export function changeStartDate(updatedDate: string | Date) {
-  const store = getStartDate();
-  store.setState({ appointmentDate: dayjs(new Date(updatedDate).setHours(0, 0, 0, 0)).format(omrsDateFormat) });
+export function changeSelectedDate(updatedDate: string | Date | dayjs.Dayjs) {
+  const store = getSelectedDate();
+  store.setState({ selectedDate: dayjs(updatedDate).hour(0).minute(0).second(0).format(omrsDateFormat) });
 }
 
-export const useAppointmentDate = () => {
-  const [currentAppointmentDate, setCurrentAppointmentDate] = useState(initialState.appointmentDate);
+export const useSelectedDate = () => {
+  const [selectedDate, setSelectedDate] = useState(initialState.selectedDate);
 
   useEffect(() => {
-    getStartDate().subscribe(({ appointmentDate }) => setCurrentAppointmentDate(appointmentDate.toString()));
+    getSelectedDate().subscribe(({ selectedDate }) => setSelectedDate(selectedDate));
   }, []);
 
-  return { currentAppointmentDate, setCurrentAppointmentDate };
+  return { selectedDate };
```

Comment:
If I'm understanding how this works, we should *never* be calling setCurrentAppointmentDate/setSelectedDate (we rather shiould be using the changeSelectedDate to update the store) so I no longer pass this back to the consumer.

Reply:
Although I literally just recommended using a React Context for this, if we _are_ using a global store, the right way to create a `use` hook for this is to use the `useStore()` hook from the framework, i.e.,

```ts
export const useSelectedDate = useStore(getSelectedDate());
```